pkg/apis/eventing/v1alpha1/eventpolicy_lifecycle_test.go

```diff
@@ -107,10 +107,115 @@ func TestEventPolicyInitializeConditions(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			test.ets.InitializeConditions()
-			if diff := cmp.Diff(test.want, test.ets, ignoreAllButTypeAndStatus); diff != "" {
+			test.eps.InitializeConditions()
+			if diff := cmp.Diff(test.want, test.eps, ignoreAllButTypeAndStatus); diff != "" {
 				t.Error("unexpected conditions (-want, +got) =", diff)
 			}
 		})
 	}
 }
+
+func TestEventPolicyReadyCondition(t *testing.T) {
+	tests := []struct {
+		name                          string
+		eps                           *EventPolicyStatus
+		markOIDCAuthenticationEnabled bool
+		markSubjectsResolvedSucceeded bool
+		wantReady                     bool
+	}{
+		{
+			name: "authenticationenabled and subjectresolved marked to true for a ready status ep",
+			eps: &EventPolicyStatus{
+				Status: duckv1.Status{
+					Conditions: []apis.Condition{
+						{Type: EventPolicyConditionReady, Status: corev1.ConditionTrue},
+						{Type: EventPolicyConditionAuthenticationEnabled, Status: corev1.ConditionTrue},
+						{Type: EventPolicyConditionSubjectsResolved, Status: corev1.ConditionTrue},
+					},
+				},
+			},
```

Comment:
Why aren't you using simply status.InitializeCondition() and do everything over the markers?

Reply:
@creydr not sure if this is the right understanding but I thought of different test cases as below and hence I initialized the `EventPolicy.Status` to Ready/NotReady instead of using `InitializeConditions`. Let me know if the tests can be written in a better way.

- `EventPolicy.Status` is Ready
     - Validate status remains true when `AuthenticationEnabled` or `SubjectsResolved` are marked to true again
     - Validate status is false when one of the `AuthenticationEnabled` or `SubjectsResolved` is marked to false 
- `EventPolicy.Status` is NotReady
     - Validate status changes to true when `AuthenticationEnabled` is previously false and marked true in the test case.
     - Validate status changes to true when  SubjectsResolved` is previously false and marked true in the test case.

Reply:
ok, than lets keep it that way. But I would then still adjust the tests `name`